### ISSUE TITLE
Nullability checks in switch variables

### ISSUE THREAD

**Issue by ZacSweers** [uber/NullAway#299]
Noticed this in our codebase this week, where nullaway was not catching this variable being dereferenced. Kotlin actually denotes this as a lint as well

![image](https://user-images.githubusercontent.com/1361086/55677121-f7f6dc00-5896-11e9-9a01-2a7a11ffaa75.png)

Example snippet:

```java
void foo(@Nullable String bar) {
  switch (bar) {  // <- should error
    default:
      break;
  }
}
```

**Comment by ZacSweers** [comment]
I've messaged @lazaroclapp the exact repro file in chat as well for a more specific repro case

**Comment by shas19** [comment]
I guess we need to add `matchSwitch` method in `Nullaway` class. Which calls existing `Nullaway::mayBeNullExpr` on non-trivial cases.
